examples/debug/carto_vl/style-guide.ipynb

```diff
@@ -0,0 +1,2382 @@
+       "\n",
+       "  carto.setDefaultAuth({\n",
+       "    username: credentials['username'],\n",
+       "    apiKey: credentials['api_key'] || 'default_public'\n",
+       "  });\n",
+       "\n",
+       "  carto.setDefaultConfig({\n",
+       "      serverURL: credentials['base_url'] || `https://${credentials['user']}.carto.com/`\n",
+       "  });\n",
+       "\n",
+       "  const sources = [{&quot;interactivity&quot;: null, &quot;is_local&quot;: false, &quot;legend&quot;: null, &quot;source&quot;: &quot;SELECT * FROM spend_data&quot;, &quot;viz&quot;: &quot;\\n            color: black\\n            strokeColor: turquoise\\n            strokeWidth: 2\\n            filter: $amount \\u003e 500\\n        &quot;}];\n",
+       "\n",
+       "  map.fitBounds([[1.6247987, 41.321587], [2.227066, 41.573303]], {animate: false});\n",
+       "\n",
+       "  \n",
+       "    map.flyTo({&quot;center&quot;: [2.17, 41.38], &quot;zoom&quot;: 12});\n",
+       "  \n",
+       "\n",
+       "  sources.forEach((elem, idx) => {\n",
+       "    let temp_source = null;\n",
+       "    if (elem.is_local) {\n",
+       "      let local_json = JSON.parse(elem.source);\n",
+       "      temp_source = new carto.source.GeoJSON(local_json);\n",
+       "    } else {\n",
+       "      temp_source = new carto.source.SQL(elem.source);\n",
+       "    }\n",
+       "\n",
+       "    const viz = new carto.Viz(elem['viz']);\n",
+       "\n",
+       "    const layer = new carto.Layer(\n",
+       "        'layer' + idx,\n",
+       "        temp_source,\n",
+       "        viz\n",
+       "    );\n",
+       "\n",
+       "    layer.addTo(map);\n",
+       "\n",
+       "    if (elem.interactivity) {\n",
+       "      const interactivity = new carto.Interactivity(layer);\n",
+       "      const tempPopup = new mapboxgl.Popup({\n",
+       "        closeButton: false,\n",
+       "        closeOnClick: false\n",
+       "      });\n",
+       "\n",
+       "      const { event, header, values} = elem.interactivity;\n",
+       "\n",
+       "      if (event === 'click') {\n",
+       "        setPopupsClick(tempPopup, interactivity, header, values);\n",
+       "      } else if (event === 'hover') {\n",
+       "        setPopupsHover(tempPopup, interactivity, header, values);\n",
+       "      }\n",
+       "    }\n",
+       "\n",
+       "    function setPopupsClick(tempPopup, interactivity, popupHeader, values) {\n",
+       "      interactivity.off('featureHover', (event) => {\n",
+       "          updatePopup(tempPopup, event, popupHeader, values)\n",
+       "      });\n",
+       "      interactivity.on('featureClick', (event) => {\n",
+       "          updatePopup(tempPopup, event, popupHeader, values)\n",
+       "      });\n",
+       "    }\n",
+       "    \n",
+       "    function setPopupsHover(tempPopup, interactivity, popupHeader, values) {\n",
+       "      interactivity.off('featureClick', (event) => {\n",
+       "       
```

Comment:
Why cols? Bear in mind that interactivity might have _variables_. I don't think we have the `cols` concept in any other place.

Why not:

```python
interactivity={
    'click': ['$prop0', '@var1'],
    'hover': ['$prop1', '@var0'],
}
```

Reply:
nice observation i like how it is explicit in the action too

Reply:
I chose `cols` because it's a great name. But your proposal is better -- I didn't realize they could be used at the same time so I made it only accept one or the other.